### ISSUE TITLE
Add environment variables to the values.yaml file

### ISSUE THREAD

**Issue by alesten** [neo4j-contrib/neo4j-helm#84]
Hey,
I have a custom plugin that I would like to run. This plugin pulls 3 variables from environment, but I can't seem to find a way to add environment variables to the helm values.yaml file. Am I missing something or is this not supported?

**Comment by moxious** [comment]
Please use custom configmaps for this: 

https://neo4j.com/labs/neo4j-helm/1.0.0/operations/#_custom_neo4j_configuration_with_configmaps

Basically, what you'll do is put whichever env vars you want into a custom configmap of your choosing, and then when you install Neo4j, you'll simply point the install to your custom configuration.

**Comment by alesten** [comment]
I had already tried this setting the `core.configMap` in my values.yaml file. But after a closer look at my configmap, I figured out that it was created wrong.

When I fix the error in my configmap, I saw the values being accessible from the environment. 

---

Another question I have then (can create a new issue if you like me to)
One of the values I need, is located in a k8s secret. Is there a way to use that as well as the configmap?

**Comment by moxious** [comment]
@alesten I don't know if you can directly pull the value of a secret into a configmap.  But I wouldn't do it even if you can....because it's a secret!  If you were able to reference a secret in a configmap, and then you push your configmap to the server, then you've exposed your secret as plaintext and what was the point of having the secret?   

There isn't a way at present to add extra secrets in the config, only a way to specify a configmap of your choosing.   But it is possible to mount a secret as an extra set of environment variables.  The helm chart right now passes the password to the container in this way.   You can see how that works here: https://github.com/neo4j-contrib/neo4j-helm/blob/master/templates/core-statefulset.yaml#L58

Basically that's just saying that part of the env can be derived from a secret.

**Comment by alesten** [comment]
Yes, that is basically what I was thinking. If I would get a secret into a configmap, I would expect it to be a reference to the secret so it wouldn't be in plaintext.

I did see that the password was available to be pulled from a secret, but as you said yourself, no way of adding extra secrets that will be put into the environment. 

So what I see is that there currently is no way to use secrets as extra environment variables, only configmaps?
Is this part of any future plans to support secrets like that, or do you stick with configmaps?

**Comment by moxious** [comment]
@alesten there's no way to do it right now.   Please provide any detail you can about why you want to do this.  If we put this into the helm chart, we'd need to know what you're trying to accomplish with it, and why a configmap isn't enough and so forth.   Otherwise we're prone to implement it maybe not in the best way

**Comment by alesten** [comment]
The use-case I have for it is as follows.
I have a custom plugin that creates a Procedure for putting data into a rabbitMQ queue. I then have an apoc trigger that for some cases call the Procedure and it continues into business logiq.
For the plugin, I need the connection for rabbitMQ. The host and username I have in the configmap, but the password I would like to keep in a secret.

I'm not 100% comfortable with k8s yet, so there might be another or better way to do what I'm trying to do.

**Comment by moxious** [comment]
@alesten please let's take this to email.  Email me at david DOT allen AT neo4j DOT com please
